src/search.rs

```diff
@@ -127,11 +125,11 @@ pub trait FindexSearch<
             let keyword = *reversed_map.get(&kwi).ok_or_else(|| {
                 Error::<CustomError>::CryptoError("Missing Kwi in reversed map.".to_string())
             })?;
-            let entry = res.entry(keyword.clone()).or_default();
-            let blocks = chain.into_iter().flat_map(|(_, v)| v).collect::<Vec<_>>();
-            for bytes in Block::unpad(&blocks)? {
-                entry.insert(IndexedValue::try_from_bytes(&bytes)?);
-            }
+            let blocks = chain
+                .into_iter()
+                .flat_map(|(_, chain_table_value)| chain_table_value.as_blocks().to_vec())
+                .collect::<Vec<_>>();
+            res.insert(keyword.clone(), IndexedValue::from_blocks(blocks)?);
```

Comment:
Same here, could `IndexedValue::from_blocks` take an iterator or something similar to avoid unnecessary copy?

Reply:
same, cf f2c5ec6